flaky TestBackoffConnector

```
=== RUN   TestBackoffConnector
      backoffconnector_test.go:96: 
          	Error Trace:	backoffconnector_test.go:96
          	Error:      	"[<swarm.Conn[*tcp.TcpTransport] /ip4/[127](https://github.com/libp2p/go-libp2p/runs/5418915605?check_suite_focus=true#step:9:127).0.0.1/tcp/33921 (12D3KooWLatVs47tAGp4QWDNRuKtGCsqWZeVjSfEV3xD7VnnJygK) <-> /ip4/127.0.0.1/tcp/42941 (12D3KooWQewg84bG8UjPKAZwAqQuJTFSnqPkLqrWZgWnF4pX2K1Y)> <swarm.Conn[*libp2pquic.transport] /ip4/127.0.0.1/udp/49984/quic (12D3KooWLatVs47tAGp4QWDNRuKtGCsqWZeVjSfEV3xD7VnnJygK) <-> /ip4/127.0.0.1/udp/42943/quic (12D3KooWGGjqT16FrJVVynvt4kV9ka4k3nDNymZeLLWm3iknMDD3)> <swarm.Conn[*tcp.TcpTransport] /ip4/127.0.0.1/tcp/33921 (12D3KooWLatVs47tAGp4QWDNRuKtGCsqWZeVjSfEV3xD7VnnJygK) <-> /ip4/127.0.0.1/tcp/43503 (12D3KooWGGjqT16FrJVVynvt4kV9ka4k3nDNymZeLLWm3iknMDD3)> <swarm.Conn[*tcp.TcpTransport] /ip4/127.0.0.1/tcp/33921 (12D3KooWLatVs47tAGp4QWDNRuKtGCsqWZeVjSfEV3xD7VnnJygK) <-> /ip4/127.0.0.1/tcp/37729 (12D3KooWLR2CJ16Ms3jcrhV5Zq7zgbfGfH35EkVzrkA8wFWQgDMb)>]" should have 3 item(s), but has 4
          	Test:       	TestBackoffConnector
          	Messages:   	wrong number of connections
  --- FAIL: TestBackoffConnector (0.91s)

--- a/p2p/discovery/backoff/backoffconnector_test.go
+++ b/p2p/discovery/backoff/backoffconnector_test.go
@@ -86,12 +86,12 @@ func TestBackoffConnector(t *testing.T) {
 	require.Eventually(t, func() bool { return len(primary.Network().Peers()) == 0 }, 3*time.Second, 10*time.Millisecond)
 
 	bc.Connect(context.Background(), loadCh(hosts))
-	require.Empty(t, primary.Network().Conns(), "shouldn't be connected to any peers")
+	require.Empty(t, primary.Network().Peers(), "shouldn't be connected to any peers")
 
 	time.Sleep(time.Millisecond * 500)
 	bc.Connect(context.Background(), loadCh(hosts))
 	require.Eventually(t, func() bool { return len(primary.Network().Peers()) == len(hosts)-2 }, 3*time.Second, 10*time.Millisecond)
 	// make sure we actually don't connect to host 1 any more
 	time.Sleep(100 * time.Millisecond)
-	require.Len(t, primary.Network().Conns(), len(hosts)-2, "wrong number of connections")
+	require.Len(t, primary.Network().Peers(), len(hosts)-2, "wrong number of connections")
 }

